python/taichi/lang/stmt_builder.py

```diff
@@ -107,85 +106,95 @@ def build_Assign(ctx, node):
         if is_static_assign:
             return node
 
-        if isinstance(node.targets[0], ast.Tuple):
-            targets = node.targets[0].elts
-
-            # Create
-            stmts = []
-
-            holder = parse_stmt('__tmp_tuple = ti.expr_init_list(0, '
-                                f'{len(targets)})')
-            holder.value.args[0] = node.value
-
-            stmts.append(holder)
-
-            def tuple_indexed(i):
-                indexing = parse_stmt('__tmp_tuple[0]')
-                StmtBuilder.set_subscript_index(indexing.value,
-                                                parse_expr("{}".format(i)))
-                return indexing.value
-
-            for i, target in enumerate(targets):
-                is_local = isinstance(target, ast.Name)
-                if is_local and ctx.is_creation(target.id):
-                    var_name = target.id
-                    target.ctx = ast.Store()
+        # Keep all generated assign statements and compose single one at last.
+        # The variable is introduced to support chained assignments.
+        # Ref https://github.com/taichi-dev/taichi/issues/2659.
+        assign_stmts = []
+        for node_target in node.targets:
+            if isinstance(node_target, ast.Tuple):
+                targets = node_target.elts
```

Comment:
nit: wrap  this part (L115-L166) in a dedicated function?

Reply:
SGTM, it should work.

Reply:
https://github.com/taichi-dev/taichi/pull/3062/files#diff-25d296ad7a362b20f650e4ccb9b456798aecaf36f7162a406f8e51db09f4255cR149 is added, I am not sure if the func name works for you, PTAL